packages/node-provider/src/node-provider.ts

```diff
@@ -39,6 +39,8 @@ export default class NodeProvider implements INodeProvider {
       return Promise.resolve(this);
     }
 
+    const context = window.parent || window;
```

Comment:
Curious why

Reply:
In order for the MessageChannel to work properly, NodeProvider needs to be frame-aware. If it can't find a parent frame, it'll fallback to try to connect to the same window.